hera_cal/lstbin.py

```diff
@@ -855,3 +961,90 @@ def sigma_clip(array, flags=None, sigma=4.0, axis=0, min_N=4):
+
+    hds : list of HERAData objects. Can have no data loaded (preferable) and should refer to single files.
+    bl_error_tol : float (meters), optional. baselines whose vector difference are within this tolerance are considered
+                   redundant. Default is 1.0 meter.
+    include_autos : bool, if True, include autos in bldicts.
+                    default is True.
+    redundant : optional, if True, each bldict stores redundant group. If False, each bldict will contain a length-1 group for each baseline
+                over all the nights.
+                default is False.
+    Outputs:
+    ---------
+    If redundant:
+        list of dictionaries of the form {0: [(a0, b0), (a0, c0)...], 1: [(a1, b1),.., ], ... Nnight: [(ANnight, BNnight), ...,]}.
+        Each dictionary represents a unique baseline length and orientation.
+        where the key of each dictionary is an index for each night to be LST binned and each value
+        is the antenna pair representing the unique baseline on that night.
+        some baseline dicts will only have a subset of nights.
+    If not redundant:
+        list of dictionaries of form {0: [(a,b)], 1:[(a, b)], 4:[(a,b)]}
+        In other words, each baseline dict corresponds to a unique baseline rather then a group
+        and each value is that baseline in a length 1 list and each key is each night in which the baseline is present.
+    """
+    # check that all hds are minimally redundant
+    blvecs = {}
+    bldicts = []
+    for night, hd in enumerate(hds):
+        assert len(hd.filepaths) == 1, 'HERAData objects must be for single data files.'
+        reds = redcal.get_reds(hd.antpos, bl_error_tol=bl_error_tol, pols=hd.pols[0], include_autos=include_autos)
+        # read to get baselines in data
+        d, _, _ = hd.read()
+        data_bls = d.antpairs()
+        reds = [[bl for bl in grp if bl[:2] in data_bls or bl[:2][::-1] in data_bls] for grp in reds]
+        reds = [grp for grp in reds if len(grp) > 0]
+        reds = [[bl[:2] for bl in grp] for grp in reds]
+        for grp in reds:
+            for bl in grp:
+                # store baseline vectors for all data.
+                blvecs[bl] = hd.antpos[bl[1]] - hd.antpos[bl[0]]
+        # otherwise, loop through baselines, for each bldict, see if the first
+        # entry matches (or conjugate matches). If yes, append to that bldict
+        for grp in reds:
+            if redundant:
+                present = False
+                for bldict in bldicts:
+                    # check if baseline group occured in previous nights
+                    # if it did, add it to the appropriate bldict.
+                    for i in bldict:
+                        if np.linalg.norm(blvecs[grp[0]] - blvecs[bldict[i][0]]) <= bl_error_tol or np.linalg.norm(blvecs[grp[0]] + blvecs[bldict[i][0]]) <= bl_error_tol:
+                            # I was having a lot of trouble getting a unittest for two separate cases here because
```

Comment:
I'd probably prefer untested code to unreadable code if the untested thing is so similar to a tested thing. I'm not a dogmatist when it comes to unit testing.

Reply:
Also, you can do something like 

```
sign = -1
if np.linalg.norm(blvecs[grp[0]] - blvecs[bldict[i][0]]) <= bl_error_tol:
    sign = 1
```

Reply:
I will use your more readable suggestion.